### ISSUE TITLE
getState()/getStateAsync() issue, if 2nd param = initial value, and 3rd param = common obj

### ISSUE THREAD

**Issue by Acgua** [ioBroker/ioBroker.javascript#924]
**Describe the bug**  

Following script from user _drefree_  is returning "not found" in console, while [documentation](https://github.com/ioBroker/ioBroker.javascript/blob/master/docs/en/javascript.md#createstate) seems to allow an initial value as second parameter, and the common object as third parameter.
```js
createState('javascript.0.variables.test', 'test init', {name: 'test Variable'}, function(){
    console.log(getState('javascript.0.variables.test').val)
})
```
When providing initial value as "def" in common, callback and async/await is working fine without errors:

```js
const state123 = '0_userdata.0.test123';
createState(state123, {name:'test variable', def:'test init', type:'string', read:true, write:true, role:'state' }, (obj) => {
    log(`Value of ${state123}: '${getState(state123).val}'`);
})
 
// Und noch Test per async/await
test();
async function test() {
 
    try {
 
        const state456 = '0_userdata.0.test456';
        await createStateAsync(state456, {name:'test variable', type:'string', read:true, write:true, role:'state', def:'test init' });
        const r = await getStateAsync(state456);
        log(`Value of ${state456}: '${r.val}'`);
 
    } catch(error) {
        log(error.stack, 'warn');
    }
 
}
```

See https://forum.iobroker.net/topic/50306/getstate-in-callback-von-createstate-gibt-not-found

**To Reproduce**  
See https://forum.iobroker.net/topic/50306/getstate-in-callback-von-createstate-gibt-not-found

**Expected behavior**  
See https://forum.iobroker.net/topic/50306/getstate-in-callback-von-createstate-gibt-not-found

**Screenshots & Logfiles**  
See https://forum.iobroker.net/topic/50306/getstate-in-callback-von-createstate-gibt-not-found

**Versions:**  
 - Adapter version: 5.2.16
 - JS-Controller version: 3.3.21
 - Node version: 14.17.1
 - Operating system: Debian

**Additional context**  
See https://forum.iobroker.net/topic/50306/getstate-in-callback-von-createstate-gibt-not-found

**Comment by Apollon77** [comment]
Yes rthis can happen and if is a timing issue ... the create will trigger a "setState" internally but the "internal state array" is only filled when the state value is published which happens slightly async. So there is a chance that in the callback the state value publish is not yet received.

In fact we would need to set the value directly in the internal array directly and do not rely on publish
